src/include/migraphx/literal.hpp

```diff
@@ -111,21 +111,7 @@ struct literal : raw_data<literal>
     void fill(Iterator start, Iterator end)
     {
         assert(std::distance(start, end) == m_shape.elements());
-        if(m_shape.standard())
-        {
-            m_shape.visit_type([&](auto as) { std::copy(start, end, as.from(buffer.get())); });
-        }
-        else
-        {
-            auto it = start;
-            m_shape.visit_type([&](auto as) {
-                auto output = make_view(m_shape, as.from(buffer.get()));
-                shape_for_each(output.get_shape(), [&](const auto& idx) {
-                    output(idx.begin(), idx.end()) = *it; // NOLINT(bugprone-signed-char-misuse)
-                    it++;
-                });
```

Comment:
From what i understand, this is not trying to make it a standard shape . Input vector from users would be provided in a standard shape. This is trying to fill the buffer such that, it puts standard shape vector input elements into non-standard shape migraphx_buffer in a correct order.  

`shape_for_each` is iterating using standard shape because input vector values are read in standard shape order. 
tensor_view `output` is assigning them into non-standard shape according to `m_shape`. tensor_view has `operator()` which calculates the correct element index using non-standard shape strides.

Reply:
I can't remember why this needed to be implemented a few years ago. But many changes have been made to the use of literals in GPU kernel implementations, so it could be fine simplifying to what Charlie has implemented. Unless certain models are failing, then we can revisit.

Reply:
thats been my apprehension since its a somewhat large block being removed with the standard case. Maybe its worth running this with DeepLearningModels library to see if we have any regressions?

Reply:
> This is trying to fill the buffer such that, it puts standard shape vector input elements into non-standard shape migraphx_buffer in a correct order.

Umang is correct here.

Reply:
> Why was literal construction made to function differently than when using a parameter?

An `argument`(which I assume is what you mean by a parameter) takes a raw buffer, not a list of values. There is a constructor to literal which will take a raw buffer as well and it will just do a simple copy:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/literal.hpp#L86

So you can convert an `argument` to a literal with `literal{arg.get_shape(), arg.data()}`.